### DISCUSSION TITLE
How can I only count unique username

### DISCUSSION THREAD

**Discussion by bunnyadad** [jertel/elastalert2#1029] [Q&A] [answered]
I wish to receive alerts that are only count by the unique username, that means ignore duplicate username.

For example, if I receive the following data points
```
{'my_data': {'username': 'alice', 'event_type': 'login'}, '@timestamp': '2016-09-20T00:00:00'}
{'my_data': {'username': 'bob', 'event_type': 'something'}, '@timestamp': '2016-09-20T00:05:00'}
{'my_data': {'username': 'alice', 'event_type': 'something else'}, '@timestamp': '2016-09-20T00:06:00'}

```
The num_events count is two, alice one and bob one. 
🙇

**Comment by nsano-rururu** [comment]
Hard to advise. I don't know how the rules are written. I can only say that I should write a rule with a rule type that can use query_key.

`query_key`
Set the field name in the Elasticsearch index to use
When set, the set field name will be aggregated as a unique key
I think it's the same as 'GROUP BY field name' in sql
https://elastalert2.readthedocs.io/en/latest/ruletypes.html#rule-types-and-configuration-options

**Comment by jertel** [comment, marked as answer]
Consider using a [metric_aggregation](https://elastalert2.readthedocs.io/en/latest/ruletypes.html#metric-aggregation) rule type:

```
type: metric_aggregation
metric_agg_type: cardinality
metric_agg_key: username
```

**Reply by bunnyadad** [reply]
Thank you, it works for me. 🙇
